eks/base/eks_managed_node_group/main.tf

```diff
@@ -0,0 +1,495 @@
+    vpc-cni = {
+      most_recent              = true
+      before_compute           = true
+      service_account_role_arn = module.vpc_cni_irsa.iam_role_arn
+      configuration_values = jsonencode({
+        env = {
+          # Reference docs https://docs.aws.amazon.com/eks/latest/userguide/cni-increase-ip-addresses.html
+          ENABLE_PREFIX_DELEGATION = "true"
+          WARM_PREFIX_TARGET       = "1"
+        }
+      })
+    }
+  }
+
+  vpc_id                   = module.vpc.vpc_id
+  subnet_ids               = module.vpc.private_subnets
+  control_plane_subnet_ids = module.vpc.intra_subnets
+
+  manage_aws_auth_configmap = true
+
+  eks_managed_node_group_defaults = {
+    ami_type                   = "AL2_x86_64"
+    instance_types             = ["m6a.2xlarge", "m6i.2xlarge", "m6a.4xlarge", "m6i.4xlarge"]
+    iam_role_attach_cni_policy = true
+  }
+
+  eks_managed_node_groups = {
+    # Default node group - as provided by AWS EKS
+    default_node_group = {
+      use_custom_launch_template = false
+
+      disk_size = 50
+
+      # Remote access cannot be specified with a launch template
+      remote_access = {
+        ec2_ssh_key               = module.key_pair.key_pair_name
+        source_security_group_ids = [aws_security_group.remote_access.id]
+      }
+    }
+
+    # Default node group - as provided by AWS EKS using Bottlerocket
+    bottlerocket_default = {
+      use_custom_launch_template = false
+
+      ami_type = "BOTTLEROCKET_x86_64"
+      platform = "bottlerocket"
+    }
+
+    # Adds to the AWS provided user data
+    bottlerocket_add = {
+      ami_type = "BOTTLEROCKET_x86_64"
+      platform = "bottlerocket"
+
+      # This will get added to what AWS provides
+      bootstrap_extra_args = <<-EOT
+        # extra args added
+        [settings.kernel]
+        lockdown = "integrity"
+      EOT
+    }
+
+    # Custom AMI, using module provided bootstrap data
+    bottlerocket_custom = {
+      # Current bottlerocket AMI
+      ami_id   = data.aws_ami.eks_default_bottlerocket.image_id
+      platform = "bottlerocket"
+
+      # Use module user data template to bootstrap
+      enable_bootstrap_user_data = true
+      # This will get added to the template
+      bootstrap_extra_args = <<-EOT
+        # The admin host container provides SSH access and runs with "superpowers".
+        # It is disabled by default, but can be disabled explicitly.
+        [settings.host-containers.admin]
+        enabled = false
+
+        # The control host container provides out-of-band access via SSM.
+        # It is enabled by default, and can be disabled if you do not expect to use SSM.
+        # This could leave you with no way to access the API and change settings on an existing node!
+        [settings.host-containers.control]
+        enabled = true
+
+        # extra args added
+        [settings.kernel]
+        lockdown = "integrity"
+
+        [settings.kubernetes.node-labels]
+        # todo: add label to the nodes
```

Comment:
can we not get them through config and not hardcode it here ?

Reply:
yeah done